sycl/test/basic_tests/memory-consumption.cpp

```diff
@@ -0,0 +1,69 @@
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: env SYCL_BE=%sycl_be %t.out
+// RUN: %t.out
+//
+//
+//==-----memory-consumption.cpp - SYCL memory consumption basic test ------==//
+//
+// This test specifically tests that memory consumption does not change
+// when the get_devices() is called repeatedly.
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "CL/sycl.hpp"
+#include <iostream>
+#include <thread>
+
+#ifdef __linux__
+#include <strings.h>
+#include <sys/resource.h>
+#include <sys/time.h>
+
+long get_cpu_mem() {
+  struct rusage usage;
+  memset(&usage, 0, sizeof(rusage));
+  getrusage(RUSAGE_SELF, &usage);
+  return usage.ru_maxrss;
+}
+#endif
+
+using namespace cl::sycl;
+
+int main() {
+  constexpr auto dev_type = info::device_type::gpu;
```

Comment:
Your test is GPU-oriented isn't it better to check all possible devices?
If we want to test all available devices we can use default device_selector and RUN lines like in https://github.com/intel/llvm/blob/sycl/sycl/test/sub_group/info.cpp#L3-L7

Reply:
> Your test is GPU-oriented isn't it better to check all possible devices?
> If we want to test all available devices we can use default device_selector and RUN lines like in https://github.com/intel/llvm/blob/sycl/sycl/test/sub_group/info.cpp#L3-L7

This test is meant to test the memory leak caused by not calling plugin APIs.
It does not make sense to test all possible devices.